File: app/src/main/java/social/entourage/android/api/tape/EncounterTapeTaskQueue.java
package social.entourage.android.api.tape;

import android.content.Context;
import android.content.Intent;

import com.squareup.tape.ObjectQueue;
import com.squareup.tape.TaskQueue;

import social.entourage.android.tour.encounter.CreateEncounterPresenter.EncounterUploadTask;

public class EncounterTapeTaskQueue extends TaskQueue<EncounterUploadTask> {

    private Context context;

    public EncounterTapeTaskQueue(ObjectQueue<EncounterUploadTask> delegate, Context context) {
        super(delegate);
        this.context = context;
    }

    public boolean start() {
        if (size() > 0) {
            context.startService(new Intent(context, EncounterTapeService.class));
            return true;
        }
        return false;
    }

    @Override
    public void add(EncounterUploadTask entry) {
        super.add(entry);
        start();
    }
}

File: app/src/main/java/social/entourage/android/api/tape/EncounterTapeTaskQueue.kt
package social.entourage.android.api.tape

import android.content.Context
import android.content.Intent
import com.squareup.tape.ObjectQueue
import com.squareup.tape.TaskQueue
import social.entourage.android.tour.encounter.CreateEncounterPresenter.EncounterUploadTask

class EncounterTapeTaskQueue(delegate: ObjectQueue<EncounterUploadTask?>?, private val context: Context) : TaskQueue<EncounterUploadTask?>(delegate) {
    fun start(): Boolean {
        if (size() > 0) {
            context.startService(Intent(context, EncounterTapeService::class.java))
            return true
        }
        return false
    }

    override fun add(entry: EncounterUploadTask?) {
        super.add(entry)
        start()
    }

}